tests/std/tests/GH_002334_branchless_clamp/test.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <string>
+
+using namespace std;
+
+int main() {
+    // Optimized cases
+    assert(clamp(2, 4, 7) == 4);
+    assert(clamp(6, 4, 7) == 6);
+    assert(clamp(8, 4, 7) == 7);
+
+    assert(clamp(2, 4, 7, std::less{}) == 4);
+    assert(clamp(6, 4, 7, std::less{}) == 6);
+    assert(clamp(8, 4, 7, std::less{}) == 7);
+
+    assert(clamp(2, 4, 7, std::less<int>{}) == 4);
+    assert(clamp(6, 4, 7, std::less<int>{}) == 6);
+    assert(clamp(8, 4, 7, std::less<int>{}) == 7);
+
+    const int data[100] = {};
+    assert(clamp(data + 22, data + 30, data + 70) == data + 30);
+    assert(clamp(data + 44, data + 30, data + 70) == data + 44);
+    assert(clamp(data + 88, data + 30, data + 70) == data + 70);
+
+    // Generic cases
+    assert(clamp(2, 4, 7, std::less<long long>{}) == 4);
+    assert(clamp(6, 4, 7, std::less<long long>{}) == 6);
+    assert(clamp(8, 4, 7, std::less<long long>{}) == 7);
+
+    assert(clamp<string>("apple", "banana", "coconut") == "banana");

```

Comment:
We shouldn't pass explicit template arguments to `clamp()` which wasn't designed for it; these should use UDLs.

Occurs below.